modules/task/execution_test.go

```diff
@@ -29,15 +29,20 @@ import (
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+	"github.com/uber-go/tally"
+	"go.uber.org/fx/service"
 )
 
 var (
-	_errorCh <-chan error
-	_ctx     = context.Background()
+	_testScope tally.Scope
+	_errorCh   <-chan error
+	_ctx       = context.Background()
 )
 
 func init() {
-	_globalBackend = NewInMemBackend()
+	host := service.NopHost()
+	_testScope = host.Metrics()
+	_globalBackend = NewInMemBackend(host)
```

Comment:
The *tests* need global state? I don't see you resetting state, so it looks like you can't make substantive assertions about these metrics. Again, these would be much less awkward if they weren't globals.